src/Disks/IDisk.h

```diff
@@ -391,6 +393,13 @@ class IDisk : public Space
     /// Required for remote disk to ensure that replica has access to data written by other node
     virtual bool checkUniqueId(const String & id) const { return exists(id); }
 
+    // TODO myrrc not sure we need a generic locking interface similar to flock/fcntl as this
+    // will be used only for remote disks with VFS support for merges and mutations.
+    // Moreover, the paths that we'd like to "lock" wouldn't be really existing ones so this looks
+    // quite hacky
+    virtual bool lock(std::string_view path, bool block);
+    virtual void unlock(std::string_view path);
+
     /// Invoked on partitions freeze query.
     virtual void onFreeze(const String &) { }
```

Comment:
maybe we can also implement something like fsync and use it for creating batches / minimizing the number of log entries

Reply:
There is a [comment](https://github.com/ClickHouse/ClickHouse/blob/ad8526cf227cbc670176a59f0098fc7db3255cb5/src/Disks/ObjectStorages/DiskObjectStorageVFSTransaction.cpp#L196) regarding performance implications of grouping